core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala

```diff
@@ -141,6 +144,43 @@ class AppStatusStoreSuite extends SparkFunSuite {
     }
   }
 
+  test("SPARK-36038: speculation summary") {
+    val store = new InMemoryStore()
+    store.write(newSpeculationSummaryData(stageId, attemptId))
```

Comment:
Keep the result of `newSpeculationStageSummary` in a variable and check against its fields in `info.foreach` below instead of hardcoding expected values.